go/types/list_test.go

```diff
@@ -958,17 +958,17 @@ func TestListDiffLargeWithSameMiddle(t *testing.T) {
 	vs1 := newLocalValueStore(cs1)
 	nums1 := generateNumbersAsValues(4000)
 	l1 := NewList(nums1...)
-	ref1 := vs1.WriteValue(l1).TargetHash()
-	vs1.Flush()
-	refList1 := vs1.ReadValue(ref1).(List)
+	hash1 := vs1.WriteValue(l1).TargetHash()
+	vs1.Flush(hash1)
```

Comment:
can you remind me why we need `Flush` for these test cases to work? It seems like reading a given value out of a ValueStore should always work if you've just written it (i.e. without flushing). I must be forgetting something....